sample-code/src/tutorials/agent-openai-langchain.ts

```diff
@@ -8,48 +8,23 @@ import {
   interrupt,
   Command
 } from '@langchain/langgraph';
-// eslint-disable-next-line import/no-internal-modules
 import { ToolNode } from '@langchain/langgraph/prebuilt';
 import { AzureOpenAiChatClient } from '@sap-ai-sdk/langchain';
 import { HumanMessage, SystemMessage } from '@langchain/core/messages';
 import { tool } from '@langchain/core/tools';
-// eslint-disable-next-line import/no-internal-modules
 import * as z from 'zod/v4';
 import type { AIMessage } from '@langchain/core/messages';
+import { getWeatherTools } from './mcp/mcp-adapter.js';
 /**
  * This example demonstrates how to create a travel itinerary assistant using LangGraph.
  * The assistant can check the weather and recommend restaurants based on the city provided.
  * It uses tools to fetch weather and restaurant data, and maintains conversation context.
  */
-const mockWeatherData: Record<
-  string,
-  { temperature: string; condition: string }
-> = {
-  paris: { temperature: '22°C', condition: 'Mild and pleasant' },
-  reykjavik: { temperature: '3°C', condition: 'Cold and windy' }
-};
-
 const mockRestaurantData: Record<string, string[]> = {
   paris: ['Le Comptoir du Relais', "L'As du Fallafel", 'Breizh Café'],
   reykjavik: ['Dill Restaurant', 'Fish Market', 'Grillmarkaðurinn']
 };
 
-const getWeatherTool = tool(
```

Comment:
[pp] I'd prefer to leave examples for both MCP and non-MCP based tools in

Reply:
Yes the other tools (mockRestraurant) is still used in the agent.